From 279807a9dd70346ee33237665934bf4db0ccec68 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Thu, 14 Jun 2018 14:49:21 +0800
Subject: [PATCH 4017/5242] MLK-18611-2 gpio: mxc: add noirq suspend/resume
 support

commit  3de10f2313473c2a5f45a40839307608a887dbdb from
https://source.codeaurora.org/external/imx/linux-imx.git

On some i.MX platforms, GPIO banks may lose power
when system suspend, and some peripheral devices
resume may need to use GPIO during resume phase
before GPIO resume, so GPIO needs to be restored
as early as possible, this patch moves GPIO
restore operation to noirq resume phase.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpio/gpio-mxc.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpio/gpio-mxc.c b/drivers/gpio/gpio-mxc.c
index d6593d8..8b03ae5 100644
--- a/drivers/gpio/gpio-mxc.c
+++ b/drivers/gpio/gpio-mxc.c
@@ -663,6 +663,7 @@ static int __maybe_unused mxc_gpio_resume(struct device *dev)
 
 static const struct dev_pm_ops mxc_gpio_dev_pm_ops = {
 	SET_SYSTEM_SLEEP_PM_OPS(mxc_gpio_suspend, mxc_gpio_resume)
+	SET_NOIRQ_SYSTEM_SLEEP_PM_OPS(mxc_gpio_runtime_suspend, mxc_gpio_runtime_resume)
 	SET_RUNTIME_PM_OPS(mxc_gpio_runtime_suspend,
 			mxc_gpio_runtime_resume, NULL)
 };
-- 
1.7.9.5

